packages/core/content-releases/server/src/controllers/release-action.ts

```diff
@@ -0,0 +1,36 @@
+import type Koa from 'koa';
+import { errors } from '@strapi/utils';
+import {
+  validateEntryContentType,
+  validateReleaseActionCreateSchema,
+  validateUniqueEntryInRelease,
+} from './validation/release-action';
+import { ReleaseActionCreateArgs, UserInfo } from '../types';
+
+const releaseActionController = {
+  async create(ctx: Koa.Context) {
+    const user: UserInfo = ctx.state.user;
+    const releaseActionArgs: ReleaseActionCreateArgs = ctx.request.body;
+
+    // Check permissions: releases can only be managed by super admins for now
+    if (!strapi.admin.services.role.hasSuperAdminRole(user)) {
+      throw new errors.ApplicationError('Content Releases is a superadmin only feature');
+    }
+
+    await Promise.all([
+      validateReleaseActionCreateSchema(releaseActionArgs),
+      validateEntryContentType(releaseActionArgs),
+      validateUniqueEntryInRelease(releaseActionArgs),
+    ]);
+
+    const releaseActionService = strapi.plugin('content-releases').service('release-action');
+    // TODO: releaseAction is of type any when the create service return is of type Promise<GetValues<"plugin::content-releases.release-action", string>> ...?
+    const releaseAction = await releaseActionService.create(releaseActionArgs);
```

Comment:
⚠️ Annoying comment alert ⚠️ This is just a personal opinion , not a change request 👼 

Imo validateEntryContentType and validateUniqueEntryInRelease is business logic related, and normally I would place those validations inside a service , rather than a controller. If there are multiple places than can create a release action, this logic would be duplicated.

And also, I would say the "action" of a release should be handled by the "release" service itself, so the entire responsibility of handling its lifecycles is on the release service, and not splited in multiple controllers (which in this case is not splitted but could be in the future).

I would rather prefer having something like :

```ts
    await validateReleaseActionCreateSchema(releaseActionArgs),

    const releaseService = strapi.plugin('content-releases').service('release');
    const releaseAction = await releaseService.createAction(releaseId, action);
```

Reply:
Also! some packages normally expose a util like [this](https://github.com/strapi/strapi/blob/ts/admin/packages/core/admin/server/src/utils/index.js) So it's easy to get the package services

```ts
const releaseAction = await getService('release').createAction(releaseId, action);
```

Reply:
I updated based on your feedback, let me know what you think :). If you have any other suggestions to improve it I'm all ears!

Reply:
looks amazing! 🚀